src/mlpack/methods/decision_tree/decision_tree_regressor_impl.hpp

```diff
@@ -1036,11 +1029,9 @@ void DecisionTreeRegressor<FitnessFunction,
   ar(CEREAL_VECTOR_POINTER(children));
 
   // Now serialize the rest of the object.
-  ar(CEREAL_NVP(splitDimension));
+  ar(CEREAL_NVP(prediction));
   ar(CEREAL_NVP(dimensionType));
-  ar(CEREAL_NVP(splitPoint));
-  // Since splitPoint and prediction are a union, we only need to serialize one of them.
```

Comment:
Ok, I updated the comment. The double serialization indeed was strange. I did not have the time to dig in to the serialization code. Do you think this was a bug that happened to work with the logic of the module, or do the test cases not exercise this code path?

Reply:
The double serialization wouldn't be a functionality bug, it would just make the serialized model something like 4 or 8 bytes too large with a bit of redundant information.  So it would still pass all the tests.  Fixing it will make the serialized `DecisionTreeRegressor`s just a bit smaller :)